detox/android/detox/src/main/java/com/wix/detox/DetoxActionHandlers.kt

```diff
@@ -119,9 +119,9 @@ class InstrumentsRecordingStateActionHandler(
 ) : DetoxActionHandler {
 
     override fun handle(params: String, messageId: Long) {
-        val recordingPath = JSONObject(params).optString("recordingPath", null)
+        val recordingPath = JSONObject(params).opt("recordingPath")
```

Comment:
```suggestion
        val recordingPath = JSONObject(params).opt("recordingPath") as String
```

Reply:
> 

This will produce crash as JSONObject$1 will not cast to String.
Just pushed more elegant code for this case.